src/audio.c

```diff
@@ -1541,17 +1541,33 @@ static int start_player(struct aurx *rx, struct audio *a)
 		prm.fmt        = rx->play_fmt;
 
 		if (!rx->aubuf) {
-			size_t psize;
-			size_t sz = aufmt_sample_size(rx->play_fmt);
+			const size_t ssz = aufmt_sample_size(rx->play_fmt);
 
-			psize = sz * calc_nsamp(prm.srate, prm.ch, prm.ptime);
+			const size_t ptime_min = a->cfg.buffer.min;
+			const size_t ptime_max = a->cfg.buffer.max;
 
-			rx->aubuf_maxsz = psize * 8;
+			size_t min_sz;
+			size_t max_sz;
 
-			err = aubuf_alloc(&rx->aubuf, psize * 1,
-					  rx->aubuf_maxsz);
-			if (err)
+			if (!ptime_min || !ptime_max)
+				return EINVAL;
+
+			min_sz = ssz*calc_nsamp(prm.srate, prm.ch, ptime_min);
```

Comment:
Maybe `min_ssz` and `max_ssz` would be more consistent to `ssz` (sample_size). Or `sz` as before.